oshi-core/src/main/java/oshi/util/GlobalConfig.java

```diff
@@ -0,0 +1,166 @@
+     * @return The property value or the given default if not found
+     */
+    public static String get(String key, String def) {
+        return configuration.getProperty(key, def);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the given default if not found
+     */
+    public static int get(String key, int def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : ParseUtil.parseIntOrDefault(value, def);
+    }
+
+    /**
+     * Get the {@code Double} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the given default if not found
+     */
+    public static Double get(String key, double def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : ParseUtil.parseDoubleOrDefault(value, def);
+    }
+
+    /**
+     * Get the {@code Boolean} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the given default if not found
+     */
+    public static boolean get(String key, boolean def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : Boolean.parseBoolean(value);
+    }
+
+    /**
+     * Set the given property, overwriting any existing value. If the given value is
+     * {@code null}, the property is removed.
+     * 
+     * @param key The property key
+     * @param val The new value
+     */
+    public static void set(String key, Object val) {
+        if (val == null)
+            configuration.remove(key);
+        else
+            configuration.setProperty(key, val.toString());
+    }
+
+    /**
+     * Reset the given property to its default value.
+     * 
+     * @param key The property key
+     */
+    public static void remove(String key) {
+        configuration.remove(key);
+    }
+
+    /**
+     * Clear the configuration.
+     */
+    public static void clear() {
+        configuration.clear();
+    }
+
+    /**
+     * Load the given {@link Properties} into the global configuration.
+     * 
+     * @param properties The new properties
+     */
+    public static void load(Properties properties) {
+        configuration.putAll(properties);
+    }
+
+    /**
+     * Indicates that a configuration value is invalid.
+     */
+    public static class PropertyException extends RuntimeException {
+
+        private static final long serialVersionUID = -7482581936621748005L;
+
+        /**
+         * @param property The property name
+         */
+        public PropertyException(String property) {
+            super("Invalid property: \"" + property + "\" = " + GlobalConfig.get(property, null));
```

Comment:
with the overloaded methods, which one does the `null` call match?   Do we really need a `null` default, as opposed to the `get()` method with no default (which returns a String or null)?

Reply:
It will match the `String` version. On this particular line, I think `get(property, null)` is acceptable, but I can bring back the `get()` without a default if you want it.

Reply:
OK, this makes sense and I'm ok with always requiring a default.